src/gcp.rs

```diff
@@ -81,8 +81,8 @@ impl<'p> GcpRef<'p> {
 impl From<&gdal_sys::GDAL_GCP> for Gcp {
     fn from(gcp: &gdal_sys::GDAL_GCP) -> Self {
         Gcp {
-            id: _string(gcp.pszId),
-            info: _string(gcp.pszId),
+            id: _string(gcp.pszId).unwrap_or_default(),
+            info: _string(gcp.pszInfo).unwrap_or_default(),
```

Comment:
:see_no_evil:

Reply:
Maybe rather replace this function with a `Gcp::from_ptr(&gdal_sys::GDAL_GCP) -> Option<Gcp>` function?

Reply:
I think it's fine, they docs say about `pszInfo`:

> Informational message or "". 

GDAL calls `json_object_new_string` on them [here](https://github.com/OSGeo/gdal/blob/0caa252cda18ce1e68028403b9719e3ca898ae50/apps/gdalinfo_lib.cpp#L806-L807), which [assumes](https://github.com/OSGeo/gdal/blob/0caa252cda18ce1e68028403b9719e3ca898ae50/ogr/ogrsf_frmts/geojson/libjson/json_object.c#L1307) that the pointer is non-null by calling `strlen` on it.

And in most places `pszInfo` is initialized with `CSLStrDup("")`, which matches the docs. As for `pszid`, that's usually initialized with a number, I don't think it can be null.